web/dev_server_client.js

```diff
@@ -0,0 +1,224 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO(yjbanov): this should be moved served by the dev server and not included in the application code.
+
+function printStats() {
+    console.log(document.querySelectorAll('*').length, 'elements');
+}
+
+class ButterflyModule {
+    constructor(moduleName, hostElement) {
+      this.moduleName = moduleName;
+      this.hostElement = hostElement;
+    }
+
+    run() {
+        let eventTypes = ["click", "keyup"];
```

Comment:
if you plan on adding more event types and want to keep track of them, I would move this to a static.  If not, I would just inline the attachments